src/sql/parse.py

```diff
@@ -217,9 +218,15 @@ def without_sql_comment(parser, line):
     return " ".join(result)
 
 
-def magic_args(magic_execute, line):
+def magic_args(magic_execute, line, cmd_from):
+    """
+    Returns the parsed arguments from the line as parsed by magic_execute
+    """
     line = without_sql_comment(parser=magic_execute.parser, line=line)
-    return magic_execute.parser.parse_args(shlex.split(line, posix=False))
+    args = shlex.split(line, posix=False)
+    if len(args) > 1:
+        check_duplicate_arguments(cmd_from, args)
```

Comment:
The code [here](https://github.com/ploomber/jupysql/blob/a1fd76c8e30c388753208dfe2ac1a1390e36dfa2/src/sql/magic_cmd.py#L74) doesn't call `parse.magic_args`. Please ensure no cases are missed

Reply:
`sqlcmd` doesn't call `parse.magic_args` which is reflected in the test for `parse.magic_args` [here](https://github.com/ploomber/jupysql/pull/908/files/c52e9976a41bbc8030985e810f30c5e8f9c84f28#diff-edd543175e806ac6654196270596c90240cc3a3cfdcba20d35d997207e4555f0R301-R339). I have not added any test for `sqlcmd`, but do have tests for both `sql` and `sqlplot` which do use the parse function.